app/src/main/java/com/github/se/travelpouch/model/activity/map/DirectionsViewModel.kt

```diff
@@ -0,0 +1,111 @@
+package com.github.se.travelpouch.model.activity.map
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.PolyUtil
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import okhttp3.OkHttpClient
+
+// Define an alias type for a path, which is a list of LatLng points
+typealias Path = List<LatLng>
+
+// Data class to hold the different route between origin and destination
+data class RouteOptions(
+    val origin: LatLng,
+    val destination: LatLng,
+    var routes: List<Path> // List of routes, where each route is a list of LatLng points
```

Comment:
Consider using `val` instead of `var` if the list routes doesn't need to be reassigned. It seems that current use does not change `routes`

Reply:
yes good catch !